pfsimulator/parflow_lib/solver_richards.c

```diff
@@ -2885,6 +2888,69 @@ AdvanceRichards(PFModule * this_module, double start_time,      /* Starting time
       }
     }
 
+
+   /***************************************************************
+    *          modify land surface pressures                      *
+    ***************************************************************/
+   
+    if (public_xtra->reset_surface_pressure == 1)
+    {
+      GrGeomSolid *gr_domain = ProblemDataGrDomain(problem_data);
+
+      int i, j, k, r, is;
+      int ix, iy, iz;
+      int nx, ny, nz;
+      int ip;
+      // JLW add declarations for use without CLM
+      Subvector *p_sub_sp;
+      double *pp_sp;
+
+      Subgrid *subgrid;
+      Grid *grid = VectorGrid(evap_trans_sum);
+
+      ForSubgridI(is, GridSubgrids(grid))
+      {
+        subgrid = GridSubgrid(grid, is);
+        p_sub_sp = VectorSubvector(instance_xtra->pressure, is);
+
+        r = SubgridRX(subgrid);
+
+        ix = SubgridIX(subgrid);
+        iy = SubgridIY(subgrid);
+        iz = SubgridIZ(subgrid);
+
+        nx = SubgridNX(subgrid);
+        ny = SubgridNY(subgrid);
+        nz = SubgridNZ(subgrid);
+
+        pp_sp = SubvectorData(p_sub_sp);
+
+        GrGeomInLoop(i, j, k, gr_domain, r, ix, iy, iz, nx, ny, nz,
+        {
+          ip = SubvectorEltIndex(p_sub_sp, i, j, k);
+          // printf(" %d %d %d %d  \n",i,j,k,ip);
+          // printf(" pp[ip] %10.3f \n",pp[ip]);
+          // printf(" NZ: %d \n",nz);
+          if (k == (nz - 1))
+          {
+            //   printf(" %d %d %d %d  \n",i,j,k,ip);
+            //   printf(" pp[ip] %10.3f \n",pp[ip]);
+
+            if (pp_sp[ip] > public_xtra->threshold_pressure)
+            {
+              printf(" pressure reset: %d %d %d %10.3f \n", i, j, k,
+                     pp_sp[ip]); pp_sp[ip] = public_xtra->reset_pressure;
+            }
+          }
+        }
+                     );
+      }
+    }
+
+    /* update pressure,  not sure if we need to do this but we might if pressures are reset along processor edges RMM */
+    handle = InitVectorUpdate(instance_xtra->pressure, VectorUpdateAll);
```

Comment:
I think the boundary update is only needed when reset_surface_pressure=1, so should be in the if {} block.

Reply:
ahh good point

Reply:
@smithsg84 - quick Q: to fix this you just mean to move the VectorUpdate inside the if{surface_pressure} block?

Reply:
Yes I think it can be moved there since we are adding for case when the threshold loop could change a value we only need to do the loop when the flag is set.

(I thought about a boolean to track if a value had been changed and only do the exchange if a threshold was found but duh that doesn't work since we don't know if changes are being made on other MPI tasks so that won't work.   I suppose we could do a reduction to see if anything changed on any task before doing the exchange.   That is likely faster if most cases don't have any thresholding.    That is a bit more complicated.    So much for quick answer to quick question :)

Reply:
thanks @smithsg84 I pushed a revised version and I think we are all set now